OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -22,49 +22,13 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
-	[Flags]
-	public enum CellConditions
-	{
-		None = 0,
-		TransientActors,
-		BlockedByMovers,
-		All = TransientActors | BlockedByMovers
-	}
-
-	public static class CellConditionsExts
-	{
-		public static bool HasCellCondition(this CellConditions c, CellConditions cellCondition)
-		{
-			// PERF: Enum.HasFlag is slower and requires allocations.
-			return (c & cellCondition) == cellCondition;
-		}
-	}
-
-	public static class CustomMovementLayerType
-	{
-		public const byte Tunnel = 1;
-		public const byte Subterranean = 2;
-		public const byte Jumpjet = 3;
-		public const byte ElevatedBridge = 4;
-	}
-
 	[Desc("Unit is able to move.")]
 	public class MobileInfo : ConditionalTraitInfo, IMoveInfo, IPositionableInfo, IFacingInfo,
 		UsesInit<FacingInit>, UsesInit<LocationInit>, UsesInit<SubCellInit>, IActorPreviewInitInfo
 	{
-		[FieldLoader.LoadUsing("LoadSpeeds", true)]
-		[Desc("Set Water: 0 for ground units and lower the value on rough terrain.")]
-		public readonly Dictionary<string, TerrainInfo> TerrainSpeeds;
-
-		[Desc("e.g. crate, wall, infantry")]
-		public readonly HashSet<string> Crushes = new HashSet<string>();
-
-		[Desc("Types of damage that are caused while crushing. Leave empty for no damage types.")]
-		public readonly HashSet<string> CrushDamageTypes = new HashSet<string>();
-
-		public readonly int WaitAverage = 5;
-
-		public readonly int WaitSpread = 2;
+		[Desc("Which Locomotor does this trait use. Must be defined on the World actor.")]
+		[LocomotorReference]
+		public readonly string Locomotor = "default";
```

Comment:
I wonder if we don't just want to `Fieldloader.Require` this and leave the default empty.